### PR TITLE
Pre-commit/prettier

### PR DESCRIPTION
**Pull request by thoragio**
Finishes #16 

* See first two commits for pre-commit/prettier set up
* See third commit for formatting changes made by prettier to current code base

### CODE REVIEW

File: source/features/user-authentication/user-authentication-reducer.js [hunk 1]
```diff
@@ -4,14 +4,14 @@ const SIGNED_IN = 'signed in';
...
 
-const actionStates = ['initial', SIGNED_OUT,
-  ['sign in', AUTHENTICATING,
-    ['report sign in failure', ERROR,
-      ['handle sign in failure', SIGNED_OUT]
-    ],
-    ['report sign in success', SIGNED_IN,
-      ['sign out', SIGNED_OUT]
-    ]
+const actionStates = [
```
Comment (ericelliott) [inline source/features/user-authentication/user-authentication-reducer.js]: Yuck. This is definitely NOT "prettier".

Comment (ericelliott) [reply source/features/user-authentication/user-authentication-reducer.js]: Is there a way to disable prettier for a short block of code? If so, we should definitely disable it for DSM state graphs.

**Comment by thoragio** [comment]
Yes, you can use `// prettier-ignore` before the block of code you don't want touched.

https://prettier.io/docs/en/ignore.html

I've reverted the DSM state graph back to the original.

**Comment by thoragio** [comment]
Because you have merged in the navbar branch to master yesterday, we will get a bunch of conflicts on this commit. Let me know if you are OK with the `prettier-ignore` option and to merge in Prettier going forward.

If so, it will be easier for me to create a new branch off of the current master and re-apply prettier instead of dealing with all of the conflicts.

**Comment by ericelliott** [comment]
Yeah, I like it. Sounds like a plan. Let's give this a try!

**Comment by thoragio** [comment]
OK. New PR on its way.

**Pull request closed without merging**